src/operator/controllers/secrets/manager.go

```diff
@@ -214,38 +113,11 @@ func (m *managerImpl) updateTLSSecretConfig(ctx context.Context, config SecretCo
 		metadata.CertTypeAnnotation:                       string(config.CertConfig.CertType),
 	}
 
-	secret.Data = secretData
+	secret.Data = certificateData.Files
```

Comment:
Assigning the secret data directly from that map means that the secret keys and values indirectly depend on what happens in other package, which doesn't have anything to do with Kubernetes secrets at all - so someone may end up changing it and breaking this inadvertently.

Instead, you should return structs that represent the two formats of secrets, and then create the secret data map explicitly here - this way, the function responsible for creating the secret is responsible for naming the secret keys, as opposed to responsibility being split between these two unrelated functions.

Another way to think about it is - can you design the function so that a reader can tell what is going to be in the secret? Currently, that requires reading another function, in another package, top to bottom.